retail/interactive-tutorial/RetailEvents.Samples/ImportUserEventsGcsSample.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright 2022 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// [START retail_import_user_events_from_gcs]
+// Import user events into a catalog from GCS using Retail API
+
+using Google.Cloud.Retail.V2;
+using Google.LongRunning;
+using System;
+
+/// <summary>
+/// The import user events GCS sample class.
+/// </summary>
+public class ImportUserEventsGcsSample
+{
+    /// <summary>
+    /// Get import user events from GCS request.
+    /// </summary>
+    /// <param name="gcsObjectName">The name of the GCS object that contains the list of user events in JSON format.</param>
+    /// <param name="projectId">The current project id.</param>
+    /// <returns>The import user events request.</returns>
+    private static ImportUserEventsRequest GetImportUserEventsGcsRequest(string gcsObjectName, string projectId, string eventsBucketName)
+    {
+        string gcsBucket = $"gs://{eventsBucketName}";
+        string gcsErrorsPrefix = $"{gcsBucket}/error";
+
+        string locationId = "global";
+        string catalogId = "default_catalog";
+        // To check error handling paste the invalid catalog name here:
+        // catalogId = "invalid_catalog_name";
+        CatalogName defaultCatalog = new CatalogName(projectId, locationId, catalogId);
+
+        GcsSource gcsSource = new GcsSource();
+        gcsSource.InputUris.Add($"{gcsBucket}/{gcsObjectName}");
```

Comment:
Collection initializers here as well.

Reply:
Here we can't do the same because `InputUris` it's readonly property.

Reply:
yes you can, you just don't initialize the collection itself, just specify the items in it.

```suggestion
        GcsSource gcsSource = new GcsSource
        {
            InputUris = { $"{gcsBucket}/{gcsObjectName}" }
        };
```